cli/config.go

```diff
@@ -4,65 +4,288 @@
 package cli
 
 import (
-	"fmt"
+	"io"
+	"net/url"
 	"os"
+	"reflect"
+	"strconv"
+	"strings"
 
 	"github.com/mainflux/mainflux/pkg/errors"
+	mfxsdk "github.com/mainflux/mainflux/pkg/sdk/go"
 	"github.com/pelletier/go-toml"
+	"github.com/spf13/cobra"
 )
 
-type Config struct {
-	Offset    uint   `toml:"offset"`
-	Limit     uint   `toml:"limit"`
+type remotes struct {
+	ThingsURL       string `toml:"things_url"`
+	UsersURL        string `toml:"users_url"`
+	ReaderURL       string `toml:"reader_url"`
+	HTTPAdapterURL  string `toml:"http_adapter_url"`
+	BootstrapURL    string `toml:"bootstrap_url"`
+	CertsURL        string `toml:"certs_url"`
+	MsgContentType  string `toml:"msg_content_type"`
+	TLSVerification bool   `toml:"tls_verification"`
+}
+
+type filter struct {
+	Offset    string `toml:"offset"`
+	Limit     string `toml:"limit"`
+	RawOutput string `toml:"raw_output"`
 	Name      string `toml:"name"`
-	RawOutput bool   `toml:"raw_output"`
+	Contact   string `toml:"contact"`
+	Email     string `toml:"email"`
+	Metadata  string `toml:"metadata"`
+}
+
+type channel struct {
+	Status string `toml:"status"`
+	State  string `toml:"state"`
+	Topic  string `toml:"topic"`
+}
+
+type config struct {
+	Remotes   remotes `toml:"remotes"`
+	Filter    filter  `toml:"filter"`
+	Channel   channel `toml:"channel"`
+	UserToken string  `toml:"user_token"`
 }
 
-// read - retrieve config from a file.
-func read(file string) (Config, error) {
-	data, err := os.ReadFile(file)
-	c := Config{}
+var (
+	errReadFail            = errors.New("failed to read config file")
+	errUnmarshalFail       = errors.New("failed to Unmarshall config TOML")
+	errConfigNotFound      = errors.New("config file was not found")
+	errUintConv            = errors.New("error converting filter to Uint64")
+	errBoolConv            = errors.New("error converting string to bool")
+	errUseExistConf        = errors.New("error using the existing configuration")
+	errNoKey               = errors.New("no such key")
+	errInvalidInt          = errors.New("error: invalid integer value for key")
+	errInvalidBool         = errors.New("error: invalid boolean value for key")
+	errUnsupportedKeyValue = errors.New("error: unsupported data type for key")
+	errMarshal             = errors.New("error marshaling the configuration")
+	errWritingConfig       = errors.New("error writing the updated config to file")
+	errInvalidURL          = errors.New("invalid url")
+	errURLParseFail        = errors.New("failed to parse url")
+	fileErr                = errors.New("file error")
+	defaultConfigPath      = "./config.toml"
```

Comment:
rename the file to this `config.toml` rather than `template.config.toml`

Reply:
not resolved

Reply:
I did not understand this, what do you mean?